src/MaplibreLayerRenderer.ts

```diff
@@ -0,0 +1,170 @@
+import {FrameState} from 'ol/Map';
+import {toDegrees} from 'ol/math';
+import {toLonLat} from 'ol/proj';
+import LayerRenderer from 'ol/renderer/Layer';
+import GeoJSON from 'ol/format/GeoJSON';
+import {Coordinate} from 'ol/coordinate';
+import {FeatureCallback} from 'ol/renderer/vector';
+import {Feature} from 'ol';
+import {Geometry, SimpleGeometry} from 'ol/geom';
+import {Pixel} from 'ol/pixel';
+import MapLibreLayer from './ol-maplibre-layer';
+
+const VECTOR_TILE_FEATURE_PROPERTY = 'vectorTileFeature';
+
+/**
+ * @private
+ */
+const formats: {
+  [key: string]: GeoJSON;
+} = {
+  'EPSG:3857': new GeoJSON({
+    featureProjection: 'EPSG:3857',
+  }),
+};
+
+/**
+ * This class is a renderer for Maplibre Layer to be able to use the native ol
+ * functionnalities like map.getFeaturesAtPixel or map.hasFeatureAtPixel.
+ * @private
+ */
+export default class MaplibreLayerRenderer extends LayerRenderer<MapLibreLayer> {
+  getFeaturesAtCoordinate(
+    coordinate: Coordinate | undefined,
+    hitTolerance: number = 5,
+  ): Feature<Geometry>[] {
+    if (!coordinate) {
+      return [];
+    }
+
+    const layer = this.getLayer();
+    const map = layer.getMapInternal();
+    const {maplibreMap} = layer;
+
+    const projection =
+      map?.getView()?.getProjection()?.getCode() || 'EPSG:3857';
+    let features: Feature[] = [];
+
+    if (!formats[projection]) {
+      formats[projection] = new GeoJSON({
+        featureProjection: projection,
+      });
+    }
+
+    if (maplibreMap) {
+      //?.isStyleLoaded()) {
+      const pixel =
+        coordinate &&
+        maplibreMap.project(toLonLat(coordinate) as [number, number]);
+
+      if (pixel?.x && pixel?.y) {
```

Comment:
pixel can be 0, no ?
And prefers early return if possible:
```suggestion
      if (pixel?.x == undefined || pixel?.y === undefined) {
        return;
      }
```